### PR TITLE
Allow users to create snapshots of their virtual machines

### PR DESCRIPTION
**Pull request by dmyerscough**
``` bash
$ salt-cloud -f create_snapshot cs name=myvm snapshot=mysnapshot description='Testing Snapshot'
[INFO    ] salt-cloud starting
[INFO    ] Creating snapshot
cs:
    ----------
    vsphere:
        ----------
        comment:
            Snapshot created
        name:
            myvm
        result:
            True
        snapshot:
            mysnapshot
```

### CODE REVIEW

File: salt/cloud/clouds/vsphere.py [hunk 1]
```diff
@@ -867,3 +867,60 @@ def snapshot_list(kwargs=None, call=None):
...
+
+def create_snapshot(kwargs=None, call=None):
+    '''
+    Create a snapshot
+
+    @name: Name of the virtual machine to snapshot
+    @snapshot: Name of the snapshot
+    @description: Description of the snapshot (optional)
+    @memory: Dump of the internal state of the virtual machine (optional)
+
```
Comment (jfindlay) [inline salt/cloud/clouds/vsphere.py]: It would be good to get a `.. versionadded:: Beryllium` here.

Comment (dmyerscough) [inline salt/cloud/clouds/vsphere.py]: I added this to the develop branch. What version should I put here?

Comment (jfindlay) [inline salt/cloud/clouds/vsphere.py]: For releases that haven't been forked from develop yet, you can tag it with the codename, which in this case is beryllium, so `.. versionadded:: Beryllium` should be sufficient, thanks.

Comment (dmyerscough) [inline salt/cloud/clouds/vsphere.py]: Done. Ill add it to another function shortly that I am writing.

Comment (jfindlay) [inline salt/cloud/clouds/vsphere.py]: Nice, thanks.

**Comment by jfindlay** [comment]
@dmyerscough, just [one lint](https://jenkins.saltstack.com/job/salt-pr-lint-n/3465/violations/file/salt/cloud/clouds/vsphere.py/).

**Pull request merged**